GLSL licences -> GNU GPL v2

sprunk opened the pull request:
Also some CC-ND -> proprietary in a handful of places, non-comprehensively

sprunk commented on luaui/Widgets/map_grass_gl4.lua:
```diff
@@ -14,7 +14,7 @@ function widget:GetInfo()
     desc      = "Instanced rendering of garbagegrass",
     author    = "Beherith (mysterme@gmail.com)",
     date      = "2021.04.12",
-    license   = "Lua code: GNU GPL, v2 or later, Shader Code: CC-BY-NC-ND 4.0",
```
ND is incompatible with GPL (since it's all about allowing derivatives).

drivver44 commented:
@Beherith i have a question
does bar org have an official email address for all licenses queries?
if yes we should get that handled in this pr

Beherith commented:
Yeah, license inquiries should go to info@beyondallreason.info
Also, I've decided to go with MIT for shaders instead of GPL, as its more permissive than GPL.

drivver44 commented:
> Also, I've decided to go with MIT for shaders instead of GPL, as its more permissive than GPL.

personally i think it makes more sense to keep that GPLv2 as the rest of the game code is licensed as GPLv2
yes the permissiveness of MIT is nice but it adds extra licenses that people need to be aware of

I personally think the less license types we have ,the better off we are. 
There was a large effort to condense the amount of license types we had in the past.

Beherith commented:
I choose MIT. Thank you for your input.

WatchTheFort commented:
@Beherith have you had a chance to give this a final review?

GoogleFrog commented:
This PR is insufficient because it lacks the text of the license. I've had a look around and there isn't a single completely consistent MIT license. Putting the full license in each file is a lot of bloat, so I propose putting it in the root (where all licenses should be anyway). Here is an example:

> The following license applies to a few files that are licensed under the MIT license, not the entire repository. 
>
> Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:
The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.

At the end of January I am going to treat BAR as if this PR has been pulled, in addition to the text above in the root. I will also tell others to treat BAR the same way.

sprunk commented:
Rebased, added text of the MIT licence, and also added text of GPLv2 since otherwise randoms who only see the MIT one could wrongly assume that is the end of the story (plus it's just required anyway).

Beherith commented:
This is still on my todo list. Please bear with me.

GoogleFrog commented:
I waited four extra months then gave up on waiting. https://github.com/ZeroK-RTS/Zero-K/commit/ca261e98d224b623a1c3a45c854244056689054f

WatchTheFort commented:
#6678 
💪😎
